pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -78,6 +78,14 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 	}
 
 	if !r.dynakube.NeedAppInjection() {
+		if r.istioReconciler != nil {
+			err = r.istioReconciler.ReconcileCodeModuleCommunicationHosts(ctx, r.dynakube)
+
+			if err != nil {
+				log.Info("Error reconciling Istio codemodules", "error", err)
+			}
+		}
+
 		return r.removeAppInjection(ctx)
```

Comment:
why are you only reconciling istio when no app injection is needed? -> this will cause to only clean up things, but never create them

you should try to reconcile in every case so we create/update when we need to or cleanup

Reply:
actually this is due to the way we do reconciliation.

I wanted to have minimal invasive changes. Before we had a short to just check if AppInjection is disabled here.

The actual reconciliation is done way down in `setupOneAgentInjection` but that never got called due to the if further up...

Reply:
moving it from `setupOneAgentInjection` to here without the is IMO less invasive than having it 2 times in different places.

Reply:
Also its checked twice now, here and in the function you check for `!r.dynakube.NeedAppInjection()`

Reply:
fixed in 94a715a4